scalable_allocator.h doesn't provide oneapi namespace

scalable_allocator.h is missing header that injects tbb namespace into oneapi namespace. This leads to compilation error if there are no other tbb includes.

```
#include <oneapi/tbb/scalable_allocator.h>
void f() {
  [[maybe_unused]] oneapi::tbb::scalable_allocator<int> x;
}
```

error C2653: 'oneapi': is not a class or namespace name

--- a/include/oneapi/tbb/scalable_allocator.h
+++ b/include/oneapi/tbb/scalable_allocator.h
@@ -1,5 +1,5 @@
 /*
-    Copyright (c) 2005-2021 Intel Corporation
+    Copyright (c) 2005-2023 Intel Corporation
 
     Licensed under the Apache License, Version 2.0 (the "License");
     you may not use this file except in compliance with the License.
@@ -20,6 +20,7 @@
 #ifdef __cplusplus
 #include "oneapi/tbb/detail/_config.h"
 #include "oneapi/tbb/detail/_utils.h"
+#include "oneapi/tbb/detail/_namespace_injection.h"
 #include <cstdlib>
 #include <utility>
 #include <new> /* std::bad_alloc() */

